README.md

```diff
@@ -1,20 +1,246 @@
+
+## Data Layer Operators
+
+If a source emits data and a destination is the recipient, what makes the two compatible?  The answer is an `operator`.  An operator is designed to perform light transformation of an object emitted from the data layer so that it can be received by a destination.  Operators are chained together using the `operators` property in a rule.  Each operator performs a small transformation and provides the result to the next operator.  An operator can also choose to not pass along data (by returning `null`) at which point data handling stops and the destination is not called.
+
+While it may take a moment to familiarize yourself with operators, they are designed to do the work that would otherwise have been custom JavaScript.  Instead of writing code, you’ll configure operators to do the work.
+
+See the [Operator Tutorial](https://github.com/fullstorydev/fullstory-data-layer-observer/tree/main/docs/operator_tutorial.md) for an in-depth guide on using a few key operators.
+
+### Operator Reference
+
+| Name | Doc Link | Description |
+| ---- | -------- | ----------- |
+| convert | Doc | Formats a value to a bool, int, real, or string. |
+| flatten | Doc | Recursively flattens all properties into an object. |
+| function | Doc | Executes a function and returns the result. |
+| insert | Doc | Inserts a value into a list at a specified position. |
+| query | Doc | Executes queries using selector syntax to return specific data within an object. |
+| suffix | Doc | Infers and appends a type suffix to an object’s properties. |
+
+Every operator requires the `name` property.  Additional options can be found by viewing an operator's documentation.
+
+> **Tip:** If an operator is required every time, use the `window['_dlo_beforeDestination']` configuration option.  This will define an operator that is always run just prior to a destination.  This can make rule writing less tedious and is applicable to scenarios like `suffix` where a `FS` destination always requires a suffixed payload.
+
+## Preview and Debugg Rules
+
+Rule writing can sometimes take a few attempts to get it right.  Fortunately, the following two options can help.
+
+- When the configuration option `_dlo_previewMode` is set to `true`, output will be written to `console.log` rather than the `destination`.
+- When a particular rule's `debug` property is set to `true`, the incremental transformations performed by `operators` and additional logging will be written to `console.debug`.
+
+Viewing the JavaScript console with the above option set to `true` prints the following example.
+
+```javascript
```

Comment:
nit: this is a little hard to read in the browser and it introduces horizontal scrolling. Would it be a good idea to add in some line breaks or make this a screenshot of the console output?

Reply:
I'm going to reformat it a bit and try to cut out some of the extraneous properties.